projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/credentials/AWSServiceAccountCredentialsService.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2023-2023 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.service.credentials;
+
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.auth.BasicSessionCredentials;
+import com.amazonaws.auth.STSAssumeRoleSessionCredentialsProvider;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenService;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClientBuilder;
+import com.amazonaws.services.securitytoken.model.AssumeRoleRequest;
+import java.util.UUID;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AWSServiceAccountCredentialsService {
+
+  @Value("${datajobs.aws.serviceAccountSecretAccessKey:}")
+  private String awsSecretAccessKey;
+
+  @Value("${datajobs.aws.serviceAccountAccessKeyId:}")
+  private String awsAccessKeyId;
+
+  @Value("${datajobs.aws.RoleArn:}")
+  private String awsRoleArn;
+
+  @Value("${datajobs.aws.defaultSessionDurationSeconds:}")
+  private int sessionDuration;
+
+  @Value("${datajobs.aws.region:}")
+  private String awsRegion;
+
+  public BasicSessionCredentials getTemporaryCredentials() {
+    AWSSecurityTokenService stsClient =
+        AWSSecurityTokenServiceClientBuilder.standard()
+            .withCredentials(
+                new AWSStaticCredentialsProvider(
+                    new BasicAWSCredentials(awsAccessKeyId, awsSecretAccessKey)))
+            .withRegion(awsRegion)
+            .build();
+
+    AssumeRoleRequest assumeRequest =
+        new AssumeRoleRequest()
+            .withRoleArn(awsRoleArn)
+            .withRoleSessionName(
+                "control-service-session-" + UUID.randomUUID().toString().substring(0, 4))
+            .withDurationSeconds(sessionDuration);
+
+    STSAssumeRoleSessionCredentialsProvider credentialsProvider =
```

Comment:
possible hint: this is likely solved by moving most of the logic here in the constructor. As by default Spring services are singleton, this class would be a single instance. 

The only gotcha could be that this method `getTemporaryCredentials` can be called by multiple threads (e.g multiple depoy request) , the aws doc say it's thread safe so should be fine.

Reply:
possible hint: this is likely solved by moving most of the logic here in the constructor. As by default Spring services are singleton, this class would be a single instance.

The only gotcha could be that this method getTemporaryCredentials can be called by multiple threads (e.g multiple depoy request) , the aws doc say it's thread safe so should be fine.

Reply:
It's not closable in the version I'm using - https://javadoc.io/doc/com.amazonaws/aws-java-sdk-sts/1.11.103/com/amazonaws/auth/STSAssumeRoleSessionCredentialsProvider.html, though I agree that it could benefit from a refactor.